python/cugraph/cugraph/gnn/comms/cugraph_nccl_comms.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+from raft_dask.common.nccl import nccl
+from raft_dask.common.comms_utils import inject_comms_on_handle_coll_only
+
+from pylibraft.common.handle import Handle
+from rmm._cuda.gpu import getDevice, setDevice
+
+from cugraph.dask.comms.comms_wrapper import init_subcomms
+
+__nccl_comms = None
+__raft_handle = None
+
+
+def nccl_init(rank: int, world_size: int, uid: int):
+    try:
+        ni = nccl()
+        ni.init(world_size, uid, rank)
+        return ni
+    except Exception as ex:
+        raise RuntimeError(f"A nccl error occurred: {ex}")
+
+
+def make_raft_handle(
+    rank, world_size, nccl_comms, n_streams_per_handle=0, verbose=False
+):
+    handle = Handle(n_streams=n_streams_per_handle)
+    inject_comms_on_handle_coll_only(handle, nccl_comms, world_size, rank, verbose)
```

Comment:
So, we don't need p2p?

Reply:
I don't think so.  As far as I can now we can run all the GNN algorithms without UCX p2p comms.

Reply:
I thought we initialized `p2p` comms in our benchmarks .

https://github.com/rapidsai/cugraph/blob/branch-24.04/benchmarks/cugraph/standalone/bulk_sampling/cugraph_bulk_sampling.py#L838


>  without UCX p2p comms.

Are we configuring UCX here? I thought this was for NCCL

Reply:
We're only using nccl here.  No UCX.  I think that's sufficient for what we're doing.

Reply:
At least that's what @ChuckHastings told me.

Reply:
OK, yeah, so p2p here means UCX p2p. NCCL has P2P as well. I think the naming here is confusing. NCCL-only might be more appropriate.